pkg/assembler/backends/arangodb/artifact.go

```diff
@@ -206,11 +206,7 @@ func (c *arangoClient) IngestArtifacts(ctx context.Context, artifacts []*model.A
 UPSERT { algorithm:doc.algorithm, digest:doc.digest } 
```

Comment:
There is an optimization we can do in the query to not use upset here as we no longer need to return the artifact. Utilizing the a composite key and using the ignore duplicates key flag.

Reply:
I can explain more but not near the computer right now.